### PR TITLE
ARROW-2755: [Python] Allow using Ninja to build extension

### PR DESCRIPTION
**Pull request by kszucs**
For slightly faster compile -> test cycles.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -231,7 +232,10 @@ def _run_cmake(self):
                 print("-- Runnning cmake for pyarrow")
                 self.spawn(cmake_command)
                 print("-- Finished cmake for pyarrow")
-                args = ['make']
+                if os.environ.get('PYARROW_CMAKE_GENERATOR') == 'Ninja':
```
Comment (pitrou) [inline python/setup.py]: Actually you should be able to use `cmake --build .` as on Windows below.

**Review by pitrou — CHANGES_REQUESTED** [review]

File: python/setup.py (hunk 1 above)
Comment (kszucs) [reply python/setup.py]: I guess then it's enough to keep the `else` branch (with minor modifications).

**Review by pitrou — APPROVED** [review]

File: python/setup.py [hunk 2]
```diff
@@ -224,46 +225,33 @@ def _run_cmake(self):
...
+            self.spawn(['cmake'] + extra_cmake_args + cmake_options + [source])
+            print("-- Finished cmake for pyarrow")
+
+            # Do the build
+            build_tool_args = []
+            if os.environ.get('PYARROW_BUILD_VERBOSE', '0') == '1':
+                build_tool_args.append('VERBOSE=1')
+            if os.environ.get('PYARROW_PARALLEL'):
+                build_tool_args.append(
+                    '-j{0}'.format(os.environ['PYARROW_PARALLEL']))
```
Comment (wesm) [inline python/setup.py]: Do these options work on Windows?

Comment (kszucs) [reply python/setup.py]: Disabled passing them. I guees I need to spin up a virtualbox sooner or later :)

**Review by wesm — APPROVED** [review]
+1, this is only failing due to the Plasma huge tables thing (and some other transient issues)

**Pull request closed without merging**